[BUG] Some .eml files fail to be saved due to naive datetime

### Description

An .eml file fails to consume.
There was the same error some time ago https://github.com/paperless-ngx/paperless-ngx/issues/2264 (solved).

---

Result message
`Dein Beleg von Epic Games.eml: Unexpected error while consuming document Dein Beleg von Epic Games.eml: localtime() cannot be applied to a naive datetime`

Input data
```json
{
  "filename": "Dein Beleg von Epic Games.eml",
  "mime_type": "message/rfc822",
  "overrides": {
    "filename": "Dein Beleg von Epic Games.eml",
    "owner_id": 3,
    "skip_asn_if_exists": false
  }
}
```

Result data
```json
{
  "error_type": "ConsumerError",
  "error_message": "Dein Beleg von Epic Games.eml: Unexpected error while consuming document Dein Beleg von Epic Games.eml: localtime() cannot be applied to a naive datetime",
  "traceback": "  File \"/usr/local/lib/python3.12/site-packages/celery/app/trace.py\", line 585, in trace_task\n    R = retval = fun(*args, **kwargs)\n                 ^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/celery/app/trace.py\", line 858, in __protected_call__\n    return self.run(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/asgiref/sync.py\", line 361, in main_wrap\n    raise exc_info[1]\n  File \"/usr/src/paperless/src/documents/tasks.py\", line 243, in consume_file\n    msg = plugin.run()\n          ^^^^^^^^^^^^\n  File \"/usr/src/paperless/src/documents/consumer.py\", line 563, in run\n    self._fail(\n  File \"/usr/src/paperless/src/documents/consumer.py\", line 243, in _fail\n    raise ConsumerError(f\"{self.filename}: {log_message or message}\") from exception\n"
}
```

### Steps to reproduce

1. Drag this email file into paperless
[Dein Beleg von Epic Games.eml](https://github.com/user-attachments/files/30342131/Dein.Beleg.von.Epic.Games.eml)
2. New task under "Needs attention"

### Webserver logs

```bash
[2026-07-24 11:54:07,419] [INFO] [paperless.consumer] [cfdad867] Consuming Dein Beleg von Epic Games.eml
[2026-07-24 11:54:08,328] [INFO] [paperless.parsing.mail] [cfdad867] Converting mail to PDF
[2026-07-24 11:54:08,466] [ERROR] [paperless.consumer] [cfdad867] Unexpected error while consuming document Dein Beleg von Epic Games.eml: localtime() cannot be applied to a naive datetime
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 361, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/consumer.py", line 520, in run
    document_parser.parse(
  File "/usr/src/paperless/src/paperless/parsers/mail.py", line 288, in parse
    self._archive_path = self.generate_pdf(mail)
                         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/paperless/parsers/mail.py", line 565, in generate_pdf
[2026-07-24 11:54:08,555] [ERROR] [paperless.tasks] [cfdad867] ConsumeTaskPlugin failed: Dein Beleg von Epic Games.eml: Unexpected error while consuming document Dein Beleg von Epic Games.eml: localtime() cannot be applied to a naive datetime
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 361, in main_wrap
  File "/usr/src/paperless/src/documents/consumer.py", line 520, in run
    document_parser.parse(
  File "/usr/src/paperless/src/paperless/parsers/mail.py", line 288, in parse
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/paperless/parsers/mail.py", line 708, in generate_pdf_from_mail
    email_html_file = self.mail_to_html(mail)
    timezone.localtime(mail.date).strftime("%Y-%m-%d %H:%M"),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/utils/timezone.py", line 183, in localtime
    raise ValueError("localtime() cannot be applied to a naive datetime")
ValueError: localtime() cannot be applied to a naive datetime
[2026-07-24 11:54:08,636] [ERROR] [celery.app.trace] Task documents.tasks.consume_file[cfdad867-c921-4f09-af17-d9c1c0c6e9cf] raised unexpected: ConsumerError('Dein Beleg von Epic Games.eml: Unexpected error while consuming document Dein Beleg von Epic Games.eml: localtime() cannot be applied to a naive datetime')
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 361, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/consumer.py", line 520, in run
    document_parser.parse(
  File "/usr/src/paperless/src/paperless/parsers/mail.py", line 288, in parse
    self._archive_path = self.generate_pdf(mail)
                         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/paperless/parsers/mail.py", line 565, in generate_pdf
    mail_pdf_file = self.generate_pdf_from_mail(mail_message)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    timezone.localtime(mail.date).strftime("%Y-%m-%d %H:%M"),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/utils/timezone.py", line 183, in localtime
    raise ValueError("localtime() cannot be applied to a naive datetime")
ValueError: localtime() cannot be applied to a naive datetime
  File "/usr/local/lib/python3.12/site-packages/celery/app/trace.py", line 585, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 361, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/tasks.py", line 243, in consume_file
          ^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/consumer.py", line 563, in run
    self._fail(
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

3.0.2

### Host OS

Synology DSM

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "3.0.2",
    "server_os": "Linux-4.4.302+-x86_64-with-glibc2.41",
    "install_type": "docker",
    "storage": {
        "total": 84830070865920,
        "available": 517176786944
    },
    "database": {
        "type": "mysql",
        "url": "paperless",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0001_squashed",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://paperless-redis:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "celery_url": "celery@9604a217a94e",
        "celery_error": null,
        "index_status": "OK",
        "index_last_modified": "2026-07-24T10:56:58.775507+02:00",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2026-07-24T09:13:48.838458Z",
        "classifier_error": null,
        "sanity_check_status": "WARNING",
        "sanity_check_last_run": null,
        "sanity_check_error": "No sanity check tasks found",
        "llmindex_status": "DISABLED",
        "llmindex_last_modified": null,
        "llmindex_error": null,
        "summary": {
            "days": 30,
            "total_count": 208,
            "pending_count": 0,
            "success_count": 205,
            "failure_count": 3
        }
    },
    "websocket_connected": "OK"
}
```

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/paperless/parsers/mail.py
+++ b/src/paperless/parsers/mail.py
@@ -272,6 +272,8 @@ class MailDocumentParser:
         logger.debug("Building formatted text from email")
         self._text = build_formatted_text(mail)
 
+        self._date = mail.date
+
         logger.debug("Creating a PDF from the email")
         if self._mailrule_id:
             rule = MailRule.objects.get(pk=self._mailrule_id)

--- a/src/paperless/parsers/mail.py
+++ b/src/paperless/parsers/mail.py
@@ -272,11 +272,6 @@ class MailDocumentParser:
         logger.debug("Building formatted text from email")
         self._text = build_formatted_text(mail)
 
-        if is_naive(mail.date):
-            self._date = make_aware(mail.date)
-        else:
-            self._date = mail.date
-
         logger.debug("Creating a PDF from the email")
         if self._mailrule_id:
             rule = MailRule.objects.get(pk=self._mailrule_id)
@@ -501,6 +496,9 @@ class MailDocumentParser:
             raise ParseError(
                 f"Could not parse {filepath}: {err}",
             ) from err
+
+        if is_naive(parsed.date):
+            parsed.date = make_aware(parsed.date)
 
         return parsed
 

--- a/src/paperless/tests/parsers/test_mail_parser.py
+++ b/src/paperless/tests/parsers/test_mail_parser.py
@@ -145,6 +145,41 @@ class TestEmailFileParsing:
         assert parsed_msg.text == "This is just a simple Text Mail.\n"
         assert parsed_msg.to == ("some@one.de",)
 
+    def test_parse_file_to_message_makes_naive_date_aware(
+        self,
+        mocker: MockerFixture,
+        mail_parser: MailDocumentParser,
+        simple_txt_email_file: Path,
+    ) -> None:
+        """
+        GIVEN:
+            - An email whose parsed date is naive (no tzinfo)
+        WHEN:
+            - The .eml file is parsed into a MailMessage
+        THEN:
+            - The resulting message date is made timezone-aware
+        """
+        mock_message = mocker.Mock(
+            from_values="mail@someserver.de",
+            date=datetime.datetime(2022, 10, 12, 21, 40, 43),
+        )
+        mocker.patch(
+            "paperless.parsers.mail.MailMessage.from_bytes",
+            return_value=mock_message,
+        )
+
+        parsed_msg = mail_parser.parse_file_to_message(simple_txt_email_file)
+
+        assert timezone.is_aware(parsed_msg.date)
+        assert parsed_msg.date.replace(tzinfo=None) == datetime.datetime(
+            2022,
+            10,
+            12,
+            21,
+            40,
+            43,
+        )
+
 
 class TestEmailMetadataExtraction:
     """

